server/sonar-server/src/main/java/org/sonar/server/platform/ServerImpl.java

```diff
@@ -48,12 +48,12 @@ public ServerImpl(Settings settings, StartupMetadata state, ServerFileSystem fs,
 
   @Override
   public String getId() {
-    return settings.getString(CoreProperties.SERVER_ID);
+    return config.get(CoreProperties.SERVER_ID).get();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Call "Optional#isPresent()" before accessing the value. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3655)